lib/logproto/logproto-proxied-text-server.c

```diff
@@ -201,9 +229,16 @@ _log_proto_proxied_text_server_handshake(LogProtoServer *s)
   LogProtoProxiedTextServer *self = (LogProtoProxiedTextServer *) s;
 
   // Fetch a line from the transport layer
-  status = log_proto_buffered_server_fetch(&self->super.super.super, &msg, &msg_len, &may_read, NULL, NULL);
-
-  self->handshake_done = (status == LPS_SUCCESS);
+  if(self->has_to_switch_to_tls)
```

Comment:
`v1_proxy_header_buff` is completely unused in case STARTTLS is not required.

We could completely remove this conditional and always use the new proxy header fetch method.
There might be 2 reasons for not using the unbuffered fetch method: performance and the fear of regression.